French translation - CSS Chapter (#1519)
HTTPArchive/almanac.httparchive.org: src/content/fr/2019/css.md

---
part_number: I
chapter_number: 2
title: CSS
description: Chapitre CSS du Web Almanac 2019, couvrant les couleurs, unités, sélecteurs, mises en page, typographies et polices, espacements, décoration, animation et media queries.
authors: [una, argyleink]
reviewers: [meyerweb, huijing]
analysts: [rviscomi]

> translators: []

Reviewer: ```suggestion
translators: [nico3333fr]
```

discuss: 1757
results: https://docs.google.com/spreadsheets/d/1uFlkuSRetjBNEhGKWpkrXo4eEIsgYelxY-qR9Pd7QpM/

> queries: 02_CSS

Reviewer: We've moved some more data into the meta data so need these translated and added here:

```suggestion
queries: 02_CSS
una_bio: Una Kravets is a Brooklyn-based international public speaker, technical writer, and Developer Advocate for Material Design at Google. Una hosts the <a href="https://www.youtube.com/watch?v=YK8GZBx3hpg">Designing the Browser</a> web series and the <a href="https://spec.fm/podcasts/toolsday">Toolsday</a> developer podcast. Follow her on <a href="https://twitter.com/una">Twitter</a> to find her musings on creative CSS, user experiences, and web development best practices.
argyleink_bio: Adam Argyle is a Google Chrome developer relations member focused on CSS; He's a web addict with an insatiable lust for great UX & UI; Find him on the web <a href="https://twitter.com/argyleink">@argyleink</a> or checkout his website <a href="https://nerdy.dev">https://nerdy.dev</a>.
featured_quote: Cascading Style Sheets (CSS) are used to paint, format, and layout web pages. Their capabilities span concepts as simple as text color to 3D perspective. It also has hooks to empower developers to handle varying screen sizes, viewing contexts, and printing. CSS helps developers wrangle content and ensure it's adapting properly to the user.
featured_stat_1: 5%
featured_stat_label_1: Pages using custom properties
featured_stat_2: 2%
featured_stat_label_2: Sites that use CSS Grid
featured_stat_3: 780
featured_stat_label_3: Number of digits in largest Z-Index value
```

published: 2019-11-11T00:00:00.000Z
last_updated: 2020-10-06T00:00:00.000Z
---

## Introduction

Les feuilles de style en cascade (CSS) sont utilisées pour peindre, mettre en forme et en page des pages Web. Leurs capacités couvrent des concepts aussi simples que la couleur du texte à la perspective 3D. Elles disposent également de possibilités pour permettre aux développeurs de gérer différentes tailles d’écran, différents contextes de visualisation, ainsi que l’impression. CSS aide les développeurs à gérer le contenu et à s’assurer qu’il s’adapte correctement à l’utilisateur.

> Lorsque vous décrivez CSS à celles et ceux qui ne sont pas familiers avec les technologies Web, il peut être utile de le considérer comme le langage pour peindre les murs de la maison&nbsp;: décrivant la taille et la position des fenêtres et des portes, ainsi que des décorations florissantes telles que du papier peint ou des plantes. La tournure amusante de cette histoire est que, selon l’utilisateur qui traverse la maison, un développeur peut adapter la maison aux préférences ou aux contextes de cet utilisateur spécifique&nbsp;!

Reviewer 2: ```suggestion
Lorsque vous décrivez CSS à celles et ceux qui ne sont pas familiers avec les technologies Web, il peut être utile de le considérer comme le langage pour peindre les murs de la maison&nbsp;: décrivant la taille et la position des fenêtres et des portes, ainsi que des décorations florissantes telles que du papier peint ou des plantes. La tournure amusante de cette histoire est que, selon la personne qui traverse la maison, un même code peut adapter la maison aux préférences ou aux contextes de cette personne spécifique&nbsp;!
```

Dans ce chapitre, nous allons inspecter, compter et extraire des données sur la façon dont CSS est utilisé sur le Web. Notre objectif est de comprendre de manière globale quelles fonctionnalités sont utilisées, comment elles sont utilisées et comment CSS se développe et est adopté.

> Prêt à fouiller dans des données fascinantes&nbsp;?! Beaucoup des nombres suivants peuvent être petits, mais ne commettez pas l’erreur de les voir comme insignifiants&nbsp;! De nombreuses années peuvent être nécessaires avant que de nouvelles choses n’envahissent le Web.

Author: Bien vu:)
J'avoue que j'ai galéré pour certaines tournures, très difficiles à traduire.

{{ figure_markup(
  image="fig2.png",
  caption="Top des couleurs nommées.",
  description="Diagramme circulaire montrant les couleurs nommées les plus populaires. Le blanc est le plus populaire à 40&nbsp;%, puis le noir à 22&nbsp;%, le rouge à 11&nbsp;% et le bleu à 5&nbsp;%.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=1985913808&format=interactive",
  width=600,
  height=415,
  data_width=600,
  data_height=415
  )
}}

La langue impacte également la couleur. Il y a plus d’exemples du «`gray`» américain que du «`grey`» britannique. Presque toutes les instances de [couleurs grises](https://www.rapidtables.com/web/color/gray-color.html) («`gray`», «`lightgray`», «`darkgray`», «`slategray`», etc.) avaient presque le double de l’utilisation en étant orthographiées avec un «a» au lieu d’un «e». Si les «`gray`»/«`grey`» étaient combinés, ils se classeraient plus haut que le bleu, en quatrième place. Cela pourrait expliquer pourquoi «`silver`» est mieux classé que «`grey`» avec un «e» dans les graphiques&nbsp;!

> 
> 

Reviewer: Extra ,line causing mismatch with English:

```suggestion

```

### Nombre de couleurs

Combien de différentes couleurs de polices sont utilisées sur le Web&nbsp;? Ce n’est donc pas le nombre total de couleurs uniques&nbsp;; c’est plutôt le nombre de couleurs différentes utilisées uniquement pour le texte. Les chiffres de ce graphique sont assez élevés et, par expérience, nous savons que sans les variables CSS, l’espacement, les tailles et les couleurs peuvent rapidement vous échapper et se fragmenter en de nombreuses valeurs minuscules dans vos styles. Ces chiffres reflètent une difficulté de gestion du style, et nous espérons que cela vous aidera à créer une perspective à rapporter à vos équipes ou projets. Comment pouvez-vous réduire ce nombre à un montant gérable et raisonnable&nbsp;?

{{ figure_markup(
  image="fig3.png",
  caption="Répartition des couleurs par page.",

>   description="202/5000
> Distribution montrant les 10e, 25e, 50e, 75e et 90e percentile de couleurs par page sur bureau et mobile. Sur le bureau, la distribution est de 8, 22, 48, 83 et 131. Les pages mobiles ont tendance à avoir plus de couleurs de 1 à 10.",

Reviewer: Unnecessary newline

```suggestion
  description="202/5000 Distribution montrant les 10e, 25e, 50e, 75e et 90e percentile de couleurs par page sur bureau et mobile. Sur le bureau, la distribution est de 8, 22, 48, 83 et 131. Les pages mobiles ont tendance à avoir plus de couleurs de 1 à 10.",
```

> Distribution montrant les 10e, 25e, 50e, 75e et 90e percentile de couleurs par page sur bureau et mobile. Sur le bureau, la distribution est de 8, 22, 48, 83 et 131. Les pages mobiles ont tendance à avoir plus de couleurs de 1 à 10.",

Author: Bien vu!

En ce qui concerne les unités basées sur l’espace physique, l’unité `cm` (centimètre) est de loin la plus populaire, suivie de `in` (pouces), puis de `Q`. Nous savons que ces types d’unités sont particulièrement utiles pour les feuilles de style d’impression, mais nous ne savions même pas que l’unité `Q` existait avant cette enquête&nbsp;! Et vous&nbsp;?

<p class="note">Une version antérieure de ce chapitre parlait de la popularité inattendue de l’unité <code>Q</code>. Grâce à la <a href="https://discuss.httparchive.org/t/chapter-2-css/1757/6">discussion de la communauté</a> concernant ce chapitre, nous avons identifié qu’il s’agissait d’un bogue dans notre analyse et avons mis à jour la figure 2.5 en conséquence.</p>

> ### Unités basées sur la taille de fenêtre (viewport)

Author: Ah cool, je ne savais pas.

### Propriétés personnalisées (custom properties)

Les propriétés personnalisées sont ce que beaucoup appellent les variables CSS. Elles sont cependant plus dynamiques qu’une variable statique classique&nbsp;! Elles sont très puissantes et en tant que communauté, nous découvrons encore leur potentiel.

{{ figure_markup(
  caption="Pourcentage de pages utilisant des propriétés personnalisées.",
  content="5&nbsp;%",
  classes="big-number"
)
}}

Nous avons eu l’impression que c’était une information passionnante, car elle montre une croissance saine de l’un de nos ajouts CSS préférés. Elles étaient disponibles dans tous les principaux navigateurs depuis 2016 ou 2017, il est donc juste de dire qu’elles sont assez récentes. De nombreuses personnes sont encore en train de passer de leurs variables de préprocesseur CSS aux propriétés personnalisées CSS. Nous estimons qu’il faudra encore quelques années avant qu’elles ne deviennent la norme.

## Sélecteurs

> ### Sélecteurs via ID vs class 

Reviewer 2: ```suggestion
### Sélecteurs via ID vs classes
```

  )
}}

Un bon tableau de suivi tel que celui qui suit, montrant que les classes occupent 93&nbsp;% des sélecteurs trouvés dans une feuille de style.

{{ figure_markup(
  image="fig8.png",
  caption="Popularité des types de sélecteurs, par sélecteur.",
  description="Diagramme à barres montrant que 94&nbsp;% des sélecteurs incluent le sélecteur de classe pour ordinateur de bureau et mobile, tandis que 7&nbsp;% des sélecteurs de bureau incluent le sélecteur par ID (8&nbsp;% pour mobile).",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=351006989&format=interactive"
  )
}}

### Sélecteurs par attributs

CSS a des sélecteurs de comparaison très puissants. Ce sont des sélecteurs comme `[target="_blank"]`, `[attribute^="value"]`, `[title~="rad"]`, `[attribute$="-rad"]` ou `[attribut*="valeur"]`. Les utilisez-vous&nbsp;? Vous pensez qu’ils sont beaucoup utilisés&nbsp;? Comparons comment ceux-ci sont utilisés avec les ID et les classes sur le Web.

{{ figure_markup(
  image="fig9.png",
  caption="Popularité des opérateurs par sélecteur d’attribut ID.",

>   description="Diagramme à barres montrant la popularité des opérateurs utilisés par les sélecteurs d’attributs d’ID. Environ 4&nbsp;% des pages de bureau et mobiles utilisent «`*=`» et «`^=`». 1&nbsp;% des pages utilisent «`=`» et «`$=`». 0&nbsp;% utilise «`~=`».", 

Reviewer: For some reason Showdown does NOT like the guillemets here and so won't generate the chapter. I can't figure out why. Do we really need them given they are in `code` styling?

```suggestion
  description="Diagramme à barres montrant la popularité des opérateurs utilisés par les sélecteurs d’attributs d’ID. Environ 4&nbsp;% des pages de bureau et mobiles utilisent `*=` et `^=`. 1&nbsp;% des pages utilisent `=` et `$=`. 0&nbsp;% utilise `~=`.", 
```

Reviewer 2: I agree, I'm not sure that the chevrons+code pattern brings any benefit. It's also used before in the chapter.

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=695879874&format=interactive"
  )
}}

{{ figure_markup(
  image="fig10.png",
  caption="Popularité des opérateurs par sélecteur d’attribut de classe.",

>   description="Diagramme à barres montrant la popularité des opérateurs utilisés par les sélecteurs d’attributs de classe. 57&nbsp;% des pages utilisent des «`*=`». 36&nbsp;% utilisent «`^=`». 1&nbsp;% utilise «`=`» et «`$=`». 0&nbsp;% utilise «`~=`».",

Reviewer: Ditto:

```suggestion
  description="Diagramme à barres montrant la popularité des opérateurs utilisés par les sélecteurs d’attributs de classe. 57&nbsp;% des pages utilisent des `*=`. 36&nbsp;% utilisent `^=`. 1&nbsp;% utilise `=` et `$=`. 0&nbsp;% utilise `~=`.",
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=377805296&format=interactive"
  )
}}

Ces opérateurs sont beaucoup plus populaires avec les sélecteurs de classe que les identifiants, ce qui semble naturel puisqu’une feuille de style a généralement moins de sélecteurs d’identifiants que de sélecteurs de classe, mais il reste agréable de voir les utilisations de toutes ces combinaisons.

### Classes par élément

Avec la montée en puissance des approches CSS comme OOCSS, <i lang="en">Atomic CSS</i> et <i lang="en">functional CSS</i> qui peuvent utiliser 10 classes ou plus sur un élément pour créer son design, peut-être verrions-nous des tendances intéressantes. La requête une fois revenue est malheureusement assez peu intéressante, la médiane sur mobile et ordinateur de bureau étant d’une classe par élément.

{{ figure_markup(
  caption="Le nombre médian de noms de classes par attribut class (bureau et mobile).",
  content="1",
  classes="big-number"
)
}}

## Disposition

> ### Flexbox

Reviewer 2: Either

```suggestion
### <i lang="en">Flexbox</i> {flexbox}
```

or

```suggestion
### `flexbox` {flexbox}
```

Reviewer 2: And same below. I prefer the use of a code wrapping myself, but an `lang` wrapping is ok too.

Reviewer 2: And same for Grid below too.

{{ figure_markup(
  caption="Pourcentage de sites Web utilisant Grid.",
  content="2&nbsp;%",
  classes="big-number"
)
}}

Cela montre à quel point la communauté de développement Web a relativement peu exercé et exploré son dernier outil de mise en page. Nous attendons avec impatience l’avènement de Grid comme moteur de mise en page principal lors de la conception d’un site. Pour nous, auteurs, nous aimons utiliser les grilles&nbsp;: nous les recherchons généralement en premier, puis nous recomposons sa complexité à mesure que nous réalisons et itérons sur la mise en page. Il reste à voir ce que le reste du monde fera de cette puissante fonctionnalité CSS au cours des prochaines années.

### Modes d’écriture

> Le Web et CSS sont des fonctionnalités de plate-forme internationale, et les modes d’écriture offrent un moyen pour HTML et CSS d’indiquer la direction de lecture et d’écriture préférées d’un utilisateur dans nos éléments.

Reviewer 2: ```suggestion
Le Web et CSS sont des fonctionnalités de plate-forme internationale, et les modes d’écriture offrent un moyen pour HTML et CSS d’indiquer la direction de lecture et d’écriture préférées d’un utilisateur ou une utilisatrice dans nos éléments.
```

{{ figure_markup(
  image="fig17.png",
  caption="Répartition du nombre de tailles de police distinctes par page.",
  description="Diagramme à barres montrant la répartition des différentes tailles de police par page. Pour les pages de bureau, les 10e, 25e, 50e, 75e et 90e centiles sont&nbsp;: 8, 20, 40, 66 et 92 comme tailles de police. La distribution de bureau diverge du mobile au 75e centile, où elle est plus grande de 7 à 13 tailles distinctes.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=1695270216&format=interactive"
  )
}}

## Espacement

### Marges

> Une marge est l’espace en dehors des éléments, comme l’espace que vous exigez lorsque vous poussez vos bras hors de vous-même. Cela ressemble souvent à l’espacement entre les éléments, mais ce n’est pas limité à cet effet. Dans un site Web ou une application, l’espacement joue un rôle important dans l’expérience utilisateur et la conception. Voyons combien de code d’espacement de marge il y a  dans une feuille de style, d’accord&nbsp;?

Reviewer 2: ```suggestion
Une marge est l’espace en dehors des éléments, comme l’espace que vous exigez lorsque vous écartez les bras. On pourrait penser que cela ne sert souvent qu'à l’espacement entre les éléments, mais ce n’est pas limité à cet effet. Dans un site Web ou une application, l’espacement joue un rôle important dans l’expérience d'utilisation et la conception. Voyons combien de code d’espacement de marge il y a  dans une feuille de style, d’accord&nbsp;?
```

{{ figure_markup(
  image="fig18.png",
  caption="Répartition du nombre de valeurs de marge distinctes par page.",
  description="Diagramme à barres montrant la distribution des valeurs de marge distinctes par page. Pour les pages sur ordinateur de  bureau, les 10e, 25e, 50e, 75e et 90e centiles sont&nbsp;: 12, 47, 96, 167 et 248 valeurs de marge distinctes. La distribution de bureau diverge du mobile au 75e centile, où elle est inférieure de 12 à 31 valeurs distinctes.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=4233531&format=interactive"
  )
}}

> C’est beaucoup, semble-t-il&nbsp;! La page de bureau médiane a 96 valeurs de marge distinctes et 104 sur mobile. Cela crée de nombreux cas d’espacements uniques dans votre conception. Curieux de savoir combien de marges vous avez sur votre site&nbsp;? Comment pouvons-nous rendre tous ces espacements plus gérables&nbsp;?

Reviewer 2: ```suggestion
C’est beaucoup, semble-t-il&nbsp;! La page de bureau médiane a 96 valeurs de marge distinctes et 104 sur mobile. Cela crée de nombreux cas d’espacements uniques dans votre conception. Nous sommes curieux de savoir&nbsp;: combien de marges avez-vous sur votre site&nbsp;? Comment pouvons-nous rendre tous ces espacements plus gérables&nbsp;?
```

> ### Propriétés logique (Logical properties)

Reviewer 2: ```suggestion
### Propriétés logiques (<i lang="en">Logical properties</i>) {proprietes-logiques-logical-properties}
```

{{ figure_markup(
  caption="Pourcentage de pages de bureau et mobiles qui incluent des propriétés logiques.",
  content="0.6&nbsp;%",
  classes="big-number"
)
}}

> Nous estimons que l’hégémonie de `margin-left` et `padding-top` sera de durée limitée, bientôt complétée par leur syntaxe de propriété logique. Bien que nous soyons optimistes, l’utilisation actuelle est assez faible à 0,67&nbsp;% d’utilisation sur les pages de bureau. Pour nous, cela ressemble à un changement d’habitude que nous devrons développer en tant qu’industrie, tout en formant, espérons-le, de nouveaux développeurs à utiliser la nouvelle syntaxe.

Reviewer 2: ```suggestion
Nous estimons que l’hégémonie de `margin-left` et `padding-top` sera de durée limitée, bientôt complétée par leur syntaxe de propriété logique. Bien que nous soyons optimistes, l’utilisation actuelle est assez faible à 0,67&nbsp;% d’utilisation sur les pages de bureau. Pour nous, cela ressemble à un changement d’habitude que nous devrons développer en tant qu’industrie, tout en formant, espérons-le, de nouveaux développeurs et développeuses à utiliser la nouvelle syntaxe.
```

## Animation

### Transitions

CSS a ce formidable pouvoir d’interpolation qui peut être simplement utilisé en écrivant une seule règle sur la façon de faire la transition de ces valeurs. Si vous utilisez CSS pour gérer les états de votre application, à quelle fréquence utilisez-vous des transitions pour effectuer la tâche&nbsp;? Interrogeons le Web&nbsp;!

{{ figure_markup(
  image="fig25.png",
  caption="Répartition du nombre de transitions par page.",
  description="Diagramme à barres montrant la répartition des transitions par page. Pour les pages de bureau, les 10e, 25e, 50e, 75e et 90e centiles sont&nbsp;: 0, 2, 16, 49 et 118 transitions. La distribution des ordinateurs de bureau est bien inférieure à celle des mobiles, avec jusqu’à 77 transitions au 90e centile.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=419145172&format=interactive"
  )
}}

> C’est plutôt bien&nbsp;! Nous avons vu `animate.css` comme une bibliothèque populaire à inclure, qui apporte une tonne d’animations de transition, mais il est toujours agréable de voir que les gens considèrent les effets de transition pour leurs interfaces utilisateur.

Reviewer 2: ```suggestion
C’est plutôt bien&nbsp;! Nous avons vu `animate.css` comme une bibliothèque populaire à inclure, qui apporte une tonne d’animations de transition, mais il est toujours agréable de voir que les gens considèrent les effets de transition pour leurs interfaces.
```

ou 

```suggestion
C’est plutôt bien&nbsp;! Nous avons vu `animate.css` comme une bibliothèque populaire à inclure, qui apporte une tonne d’animations de transition, mais il est toujours agréable de voir que les gens considèrent les effets de transition pour leurs interfaces d'utilisation.
```

### Animations d’images clés

Les animations d’images clés (<i lang="en">keyframe</i>) CSS sont une excellente solution pour vos animations ou transitions plus complexes. Ils vous permettent d’être plus explicite, ce qui offre un meilleur contrôle sur les effets. Ils peuvent être petits, comme un unique effet, ou être plus complexes avec de nombreux effets composés via <i lang="en">keyframe</i> dans une animation plus riche. Le nombre médian d’animations via <i lang="en">keyframe</i> par page est bien inférieur à celui des transitions CSS.

{{ figure_markup(
  image="fig26.png",
  caption="Répartition du nombre d’images clés par page.",
  description="Diagramme à barres montrant la répartition des images clés par page. Pour les pages mobiles, les 10e, 25e, 50e, 75e et 90e centiles sont&nbsp;: 0, 0, 3, 18 et 76 images clés. La distribution mobile est légèrement supérieure à celle des ordinateurs de bureau, de 6 images clés aux 75e et 90e percentiles.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=11848448&format=interactive"
  )
}}

> ## Requête de media (media queries)

Reviewer 2: ```suggestion
## Requêtes de media (<i lang="en">media queries</i>) {requetes-de-media-queries}
```

Les <i lang="en">media queries</i> permettent à CSS de se connecter à diverses variables au niveau du système afin de s’adapter de manière appropriée à l’utilisation du visiteur. Certaines de ces <i lang="en">media queries</i> peuvent gérer les styles d’impression, les styles d’écran du projecteur et la taille de la fenêtre/de l’écran. Pendant longtemps, les <i lang="en">media queries</i> ont été principalement exploitées pour leur connaissance de la fenêtre d’affichage. Les concepteurs et les développeurs pouvaient adapter leurs mises en page aux petits écrans, aux grands écrans, etc. Plus tard, le Web a commencé à apporter de plus en plus de fonctionnalités et de possibilités, ce qui signifie que les <i lang="en">media queries</i> peuvent désormais gérer les fonctionnalités d’accessibilité en plus des fonctionnalités de la fenêtre.

Un bon point de départ avec les <i lang="en">media queries</i>&nbsp;: combien sont utilisées par page&nbsp;? À combien de moments ou de contextes différents la page type a-t-elle envie de satisfaire&nbsp;?

{{ figure_markup(
  image="fig27.png",

>   caption="Répartition du nombre de <i lang="en">media queries</i> par page.",
>   description="Diagramme à barres montrant la répartition des <i lang="en">media queries</i> par page. Pour les pages de bureau, les 10e, 25e, 50e, 75e et 90e centiles sont&nbsp;: 0, 3, 14, 27 et 43 media-queries. La distribution de bureau est similaire à la distribution mobile.",

Reviewer: As you are using `lang="en"` you must use single quotes:

```suggestion
  caption='Répartition du nombre de <i lang="en">media queries</i> par page.',
  description='Diagramme à barres montrant la répartition des <i lang="en">media queries</i> par page. Pour les pages de bureau, les 10e, 25e, 50e, 75e et 90e centiles sont&nbsp;: 0, 3, 14, 27 et 43 media-queries. La distribution de bureau est similaire à la distribution mobile.',
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=1892465673&format=interactive"
  )
}}

### Tailles populaire des points d’arrêt (breakpoints) des media queries

Pour les <i lang="en">media queries</i> basées sur la fenêtre d’affichage, tout type d’unité CSS peut être passé dans l’expression de requête pour évaluation. Autrefois, les gens passaient `em` et `px` dans la requête, mais plus d’unités ont été ajoutées au fil du temps, ce qui nous rendait très curieux de savoir quels types de tailles étaient couramment trouvables sur le Web. Nous supposons que la plupart des <i lang="en">media queries</i> suivront les tailles d’appareils courants, mais au lieu de supposer, examinons les données&nbsp;!

{{ figure_markup(
  image="fig28.png",

>   caption="Valeurs les plus fréquemment utilisées dans les <i lang="en">media queries</i>.",
>   description="Diagramme à barres des valeurs les plus courantes utilisées avec les <i lang="en">media queries</i>. 768px et 767px sont respectivement les plus populaires à 23&nbsp;% et 16&nbsp;%. La liste tombe rapidement après cela, avec 992px utilisé 6&nbsp;% du temps et 1200px utilisé 4&nbsp;% du temps. Le bureau et le mobile ont une utilisation similaire.",

Reviewer: ```suggestion
  caption='Valeurs les plus fréquemment utilisées dans les <i lang="en">media queries</i>.',
  description='Diagramme à barres des valeurs les plus courantes utilisées avec les <i lang="en">media queries</i>. 768px et 767px sont respectivement les plus populaires à 23&nbsp;% et 16&nbsp;%. La liste tombe rapidement après cela, avec 992px utilisé 6&nbsp;% du temps et 1200px utilisé 4&nbsp;% du temps. Le bureau et le mobile ont une utilisation similaire.',
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=1353707515&format=interactive"
  )
}}

La figure 2.28 ci-dessus montre qu’une partie de nos hypothèses était correcte&nbsp;: il y a certainement une grande quantité de tailles spécifiques aux téléphones, mais il y en a aussi qui ne le sont pas. Il est également intéressant de voir comment les pixels sont très dominants, avec quelques entrées utilisant `em` au-delà de la portée de ce graphique.

### Utilisation du portrait vs paysage

La valeur la plus populaire semble être `768px`, ce qui a attisé notre curiosité. Cette valeur a-t-elle été principalement utilisée pour passer à une mise en page portrait, car elle pourrait être basée sur l’hypothèse que `768px` représente la fenêtre de portrait mobile typique&nbsp;? Nous avons donc lancé une requête pour voir la popularité de l’utilisation des modes portrait et paysage&nbsp;:

{{ figure_markup(
  image="fig29.png",

>   caption="Adoption des modes d’orientation des <i lang="en">media queries</i>.",
>   description="Diagramme à barres montrant l’adoption des modes d’orientation portrait et paysage des <i lang="en">media queries</i>. 31&nbsp;% des pages spécifient paysage, 8&nbsp;% spécifient portrait et 7&nbsp;% spécifient les deux. L’adoption est la même pour les pages de bureau et mobiles.",

Reviewer: ```suggestion
  caption='Adoption des modes d’orientation des <i lang="en">media queries</i>.',
  description='Diagramme à barres montrant l’adoption des modes d’orientation portrait et paysage des <i lang="en">media queries</i>. 31&nbsp;% des pages spécifient paysage, 8&nbsp;% spécifient portrait et 7&nbsp;% spécifient les deux. L’adoption est la même pour les pages de bureau et mobiles.',
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=295845630&format=interactive"
  )
}}

Fait intéressant, `portrait` n’est pas beaucoup utilisé, alors que `landscape` l’est beaucoup plus. Nous ne pouvons que supposer que `768px` a été suffisamment fiable comme cas de mise en page de portrait pour être raisonnablement atteint. Nous supposons également que les utilisateurs d’un ordinateur de bureau, qui testent leur travail, ne peuvent pas déclencher le portrait pour voir leur disposition mobile aussi facilement qu’ils peuvent simplement réduire le navigateur. Difficile à dire, mais les données sont fascinantes.

### Types d’unités les plus populaires

Dans les <i lang="en">media queries</i> de largeur et de hauteur que nous avons vues jusqu’à présent, les pixels semblent être l’unité dominante de choix pour les développeurs qui cherchent à adapter leur interface utilisateur aux fenêtres. Nous voulions cependant questionner plus avant ce point et jeter un coup d’œil aux types d’unités que les gens utilisent. Voici ce que nous avons trouvé.

{{ figure_markup(
  image="fig30.png",

>   caption="Adoption d’unités dans les <i lang="en">breakpoints</i> des <i lang="en">media queries</i>.",
>   description="Diagramme à barres montrant 75&nbsp;% <i lang="en">breakpoints</i> des <i lang="en">media queries</i> spécifiant les pixels, 8&nbsp;% spécifiant ems et 1&nbsp;% spécifiant les rems.",

Reviewer: ```suggestion
  caption='Adoption d’unités dans les <i lang="en">breakpoints</i> des <i lang="en">media queries</i>.',
  description='Diagramme à barres montrant 75&nbsp;% <i lang="en">breakpoints</i> des <i lang="en">media queries</i> spécifiant les pixels, 8&nbsp;% spécifiant ems et 1&nbsp;% spécifiant les rems.',
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=305563768&format=interactive"
  )
}}

### `min-width` vs `max-width`

Lorsque les gens écrivent une <i lang="en">media query</i>, visent-ils généralement une taille de fenêtre située au-dessus ou en dessous d’une plage spécifique, _ou_ les deux, ou encore si elle se situe entre une plage de tailles&nbsp;? Demandons au web&nbsp;!

{{ figure_markup(
  image="fig31.png",

>   caption="Adoption des propriétés utilisées dans les <i lang="en">breakpoints</i> des <i lang="en">media queries</i>",

Reviewer: ```suggestion
  caption='Adoption des propriétés utilisées dans les <i lang="en">breakpoints</i> des <i lang="en">media queries</i>',
```

  description="Diagramme à barres montrant que 74&nbsp;% des pages de bureau utilisent `max-width`, 70&nbsp;% `min-width` et 68&nbsp;% utilisant les deux propriétés. L’adoption est similaire pour les pages mobiles.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=2091525146&format=interactive"
  )
}}

Aucun gagnant clair ici, `max-width` et `min-width` sont presque à égalité.

### Impression et vocalisation

Les sites Web ressemblent à du papier numérique, non&nbsp;? En tant qu’utilisateurs et utilisatrices, il est généralement connu que vous pouvez simplement lancer l’impression à partir de votre navigateur et transformer ce contenu numérique en contenu physique. Un site Web n’est pas obligé de se changer pour ce cas d’utilisation, mais il le peut s’il le souhaite&nbsp;! Moins connue est la possibilité d’ajuster votre site Web dans le cas où il est lu par un outil ou un robot. Alors, à quelle fréquence ces fonctionnalités sont-elles exploitées&nbsp;?

{{ figure_markup(
  image="fig32.png",

>   caption="Adoption des medias `all`, `print`, `screen` et `speech` pour les <i lang="en">media queries</i>.",

Reviewer: ```suggestion
  caption='Adoption des medias `all`, `print`, `screen` et `speech` pour les <i lang="en">media queries</i>.',
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=2132635319&format=interactive"
  )
}}

Voir le chapitre [Poids des page](./page-weight) pour un aperçu plus détaillé du nombre d’octets que les sites Web chargent pour chaque type de contenu.

### Bibliothèques

Il est commun, populaire, pratique et puissant d’utiliser une bibliothèque CSS pour lancer un nouveau projet. Bien que vous ne soyez peut-être pas du genre à rechercher une bibliothèque, nous avons interrogé le Web en 2019 pour voir lesquelles sont en tête du peloton. Si les résultats vous étonnent, comme pour nous, c’est un indice intéressant de la petite taille d’une bulle dans laquelle nous vivons. Les choses peuvent sembler extrêmement populaires, mais lorsque le Web est réellement interrogé, la réalité est un peu différente.

<figure>
  <table>
    <thead>
      <tr>
        <th scope="col">Bibliothèque</th>
        <th scope="col">Bureau</th>
        <th scope="col">Mobile</th>
      </tr>
    </thead>
    <tbody>
      <tr>

>         <td>Bootstrap</td>

Reviewer 2: ```suggestion
        <td lang="en">Bootstrap</td>
```

        <td class="numeric">27.8&nbsp;%</td>
        <td class="numeric">26.9&nbsp;%</td>
      </tr>
      <tr>

>         <td>animate.css</td>

Reviewer 2: ```suggestion
        <td lang="en">animate.css</td>
```

        <td class="numeric">6.1&nbsp;%</td>
        <td class="numeric">6.4&nbsp;%</td>
      </tr>
      <tr>

>         <td>ZURB Foundation</td>

Reviewer 2: ```suggestion
        <td lang="en">ZURB Foundation</td>
```

        <td class="numeric">2.5&nbsp;%</td>
        <td class="numeric">2.6&nbsp;%</td>
      </tr>
      <tr>

>         <td>UIKit</td>

Reviewer 2: ```suggestion
        <td lang="en">UIKit</td>
```

        <td class="numeric">0.5&nbsp;%</td>
        <td class="numeric">0.6&nbsp;%</td>
      </tr>
      <tr>

>         <td>Material Design Lite</td>

Reviewer 2: ```suggestion
        <td lang="en">Material Design Lite</td>
```

        <td class="numeric">0.3&nbsp;%</td>
        <td class="numeric">0.3&nbsp;%</td>
      </tr>
      <tr>

>         <td>Materialize CSS</td>

Reviewer 2: ```suggestion
        <td lang="en">Materialize CSS</td>
```

        <td class="numeric">0.2&nbsp;%</td>
        <td class="numeric">0.2&nbsp;%</td>
      </tr>
      <tr>

>         <td>Pure CSS</td>

Reviewer 2: ```suggestion
        <td lang="en">Pure CSS</td>
```

        <td class="numeric">0.1&nbsp;%</td>
        <td class="numeric">0.1&nbsp;%</td>
      </tr>
      <tr>

>         <td>Angular Material</td>

Reviewer 2: ```suggestion
        <td lang="en">Angular Material</td>
```

        <td class="numeric">0.1&nbsp;%</td>
        <td class="numeric">0.1&nbsp;%</td>
      </tr>
      <tr>

>         <td>Semantic-ui</td>

Reviewer 2: ```suggestion
        <td lang="en">Semantic-ui</td>
```

        <td class="numeric">0.1&nbsp;%</td>
        <td class="numeric">0.1&nbsp;%</td>
      </tr>
      <tr>

>         <td>Bulma</td>

Reviewer 2: ```suggestion
        <td lang="en">Bulma</td>
```

        <td class="numeric">0.0&nbsp;%</td>
        <td class="numeric">0.0&nbsp;%</td>
      </tr>
      <tr>

>         <td>Ant Design</td>

Reviewer 2: ```suggestion
        <td lang="en">Ant Design</td>
```

        <td class="numeric">0.0&nbsp;%</td>
        <td class="numeric">0.0&nbsp;%</td>
      </tr>
      <tr>

>         <td>tailwindcss</td>

Reviewer 2: ```suggestion
        <td lang="en">tailwindcss</td>
```

        <td class="numeric">0.0&nbsp;%</td>
        <td class="numeric">0.0&nbsp;%</td>
      </tr>
      <tr>

>         <td>Milligram</td>

Reviewer 2: ```suggestion
        <td lang="en">Milligram</td>
```

        <td class="numeric">0.0&nbsp;%</td>
        <td class="numeric">0.0&nbsp;%</td>
      </tr>
      <tr>

>         <td>Clarity</td>

Reviewer 2: ```suggestion
        <td lang="en">Clarity</td>
```

        <td class="numeric">0.0&nbsp;%</td>
        <td class="numeric">0.0&nbsp;%</td>
      </tr>
    </tbody>
  </table>
  <figcaption>{{ figure_link(caption="Pourcentage de pages qui incluent une bibliothèque CSS donnée.") }}</figcaption>
</figure>

> Ce graphique suggère que [Bootstrap](https://getbootstrap.com/) est une bibliothèque précieuse à connaître pour vous aider à trouver un emploi. Regardez toutes les opportunités qui vous sont offertes&nbsp;! Il convient également de noter qu’il ne s’agit que d’un signal positif&nbsp;: les calculs ne totalisent pas 100&nbsp;% car tous les sites n’utilisent pas de bibliothèque CSS. Un peu plus de la moitié de tous les sites _n’utilisent pas_ de bibliothèque CSS connue. Très intéressant, non&nbsp;?!

Reviewer 2: ```suggestion
Ce graphique suggère que [<span lang="en">Bootstrap</span>](https://getbootstrap.com/) est une bibliothèque précieuse à connaître pour vous aider à trouver un emploi. Regardez toutes les opportunités qui vous sont offertes&nbsp;! Il convient également de noter qu’il ne s’agit que d’un signal positif&nbsp;: les calculs ne totalisent pas 100&nbsp;% car tous les sites n’utilisent pas de bibliothèque CSS. Un peu plus de la moitié de tous les sites _n’utilisent pas_ de bibliothèque CSS connue. Très intéressant, non&nbsp;?!
```

### Resets CSS

Les <i lang="en">resets</i> CSS ont l’intention de normaliser ou de créer une base de référence pour les éléments Web natifs. Au cas où vous ne le sauriez pas, chaque navigateur propose sa propre feuille de style pour tous les éléments HTML, et chaque navigateur prend ses propres décisions sur l’apparence ou le comportement de ces éléments. Les <i lang="en">resets</i> ont examiné ces fichiers, trouvé leur terrain d’entente (ou non) et aplani toutes les différences afin que vous, en tant que développeur ou développeuse, puissiez styliser dans un navigateur et avoir une confiance raisonnable que l’apparence soit la même dans un autre.

Voyons donc combien de sites en utilisent un&nbsp;! Leur existence semble tout à fait raisonnable, alors combien de personnes sont d’accord avec leurs choix et les utilisent sur leurs sites&nbsp;?

{{ figure_markup(
  image="fig37.png",

>   caption="Adoption des <i lang="en">resets</i> CSS.",
>   description="Diagramme à barres montrant l’adoption de trois <i lang="en">resets</i> CSS&nbsp;: Normalize.css (33&nbsp;%), Reset CSS (3&nbsp;%) et Pure CSS (0&nbsp;%). Il n’y a aucune différence d’adoption entre les pages de bureau et mobiles.",

Reviewer: ```suggestion
  caption='Adoption des <i lang="en">resets</i> CSS.',
  description='Diagramme à barres montrant l’adoption de trois <i lang="en">resets</i> CSS&nbsp;: Normalize.css (33&nbsp;%), Reset CSS (3&nbsp;%) et Pure CSS (0&nbsp;%). Il n’y a aucune différence d’adoption entre les pages de bureau et mobiles.',
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=1318910215&format=interactive"
  )
}}

> Il s’avère qu’environ un tiers du Web utilise [`normalize.css`](https://necolas.github.io/normalize.css), ce qui pourrait être considéré comme une approche douce, en tout cas plus que celle d’un pur <i lang="en">reset</i> CSS. Nous avons regardé un peu plus en profondeur, et il s’avère que Bootstrap inclut `normalize.css`, ce qui représente probablement une grande partie de son utilisation. Il convient également de noter que `normalize.css` a plus d’adoption que Bootstrap, donc il y a beaucoup de gens qui l’utilisent seul.

Reviewer 2: ```suggestion
Il s’avère qu’environ un tiers du Web utilise [`normalize.css`](https://necolas.github.io/normalize.css), ce qui pourrait être considéré comme une approche douce, en tout cas plus que celle d’un pur <i lang="en">reset</i> CSS. Nous avons regardé un peu plus en profondeur, et il s’avère que Bootstrap inclut `normalize.css`, ce qui représente probablement une grande partie de son utilisation. Il convient également de noter que `normalize.css` a plus d’adoption que Bootstrap, donc il y a beaucoup de gens qui l’utilisent seul.
```
```suggestion
Il s’avère qu’environ un tiers du Web utilise [`normalize.css`](https://necolas.github.io/normalize.css), ce qui pourrait être considéré comme une approche douce, en tout cas plus que celle d’un pur <i lang="en">reset</i> CSS. Nous avons regardé un peu plus en profondeur, et il s’avère que <span lang="en">Bootstrap</span> inclut `normalize.css`, ce qui représente probablement une grande partie de son utilisation. Il convient également de noter que `normalize.css` a plus d’adoption que <span lang="en">Bootstrap</span>, donc il y a beaucoup de gens qui l’utilisent seul.
```

### `@supports` et `@import`

CSS `@supports` est un moyen pour le navigateur de vérifier si une combinaison propriété-valeur particulière est analysée comme valide, puis d’appliquer des styles si la vérification est vraie.

{{ figure_markup(
  image="fig38.png",

>   caption=’Popularité des règles CSS «@».’,
>   description=’Diagramme à barres montrant la popularité des règles `@import` et `@supports`. Sur le bureau, `@import` est utilisé sur 28&nbsp;% des pages et `@supports` sur 31&nbsp;%. Pour mobile, `@import` est utilisé sur 26&nbsp;% des pages et `@supports` sur 29&nbsp;%’,

Reviewer: Wrong quotes here:

```suggestion
  caption='Popularité des règles CSS «@».",
  description="Diagramme à barres montrant la popularité des règles `@import` et `@supports`. Sur le bureau, `@import` est utilisé sur 28&nbsp;% des pages et `@supports` sur 31&nbsp;%. Pour mobile, `@import` est utilisé sur 26&nbsp;% des pages et `@supports` sur 29&nbsp;%",
```

---
part_number: I
chapter_number: 2
title: CSS
description: Chapitre CSS du Web Almanac 2019, couvrant les couleurs, unités, sélecteurs, mises en page, typographies et polices, espacements, décoration, animation et media queries.
authors: [una, argyleink]
reviewers: [meyerweb, huijing]
analysts: [rviscomi]
translators: [nico3333fr]
discuss: 1757
results: https://docs.google.com/spreadsheets/d/1uFlkuSRetjBNEhGKWpkrXo4eEIsgYelxY-qR9Pd7QpM/
queries: 02_CSS

> una_bio: Una Kravets is a Brooklyn-based international public speaker, technical writer, and Developer Advocate for Material Design at Google. Una hosts the <a href="https://www.youtube.com/watch?v=YK8GZBx3hpg">Designing the Browser</a> web series and the <a href="https://spec.fm/podcasts/toolsday">Toolsday</a> developer podcast. Follow her on <a href="https://twitter.com/una">Twitter</a> to find her musings on creative CSS, user experiences, and web development best practices.

Reviewer 2: ```suggestion
una_bio: Una Kravets est une conférencière internationale basée à Brooklyn, rédactrice technique et <i lang="en">Developer Advocate</i> pour <i lang="en">Material Design</i> chez Google. Una anime la série web <a href="https://www.youtube.com/watch?v=YK8GZBx3hpg" lang="en">Designing the Browser</a> et le podcast <a href="https://spec.fm/podcasts/toolsday" lang="en">Toolsday</a>developpeur. Suivez-la sur <a href="https://twitter.com/una">Twitter</a> pour découvrir ses réflexions sur les CSS créatifs, les expériences utilisateurs et les meilleures pratiques de développement web.
```

> argyleink_bio: Adam Argyle is a Google Chrome developer relations member focused on CSS; He's a web addict with an insatiable lust for great UX & UI; Find him on the web <a href="https://twitter.com/argyleink">@argyleink</a> or checkout his website <a href="https://nerdy.dev">https://nerdy.dev</a>.

Reviewer 2: ```suggestion
argyleink_bio: Adam Argyle est un membre de l’équipe  <i lang="en">Google Chrome Developer Relations</i>, spécialisé dans le CSS. C’est un accro du web avec une soif insatiable de super UX & UI. Retrouvez-le sur le web <a href="https://twitter.com/argyleink">@argyleink</a> ou consultez son site <a href="https://nerdy.dev">https://nerdy.dev</a>.
```

> featured_quote: Cascading Style Sheets (CSS) are used to paint, format, and layout web pages. Their capabilities span concepts as simple as text color to 3D perspective. It also has hooks to empower developers to handle varying screen sizes, viewing contexts, and printing. CSS helps developers wrangle content and ensure it's adapting properly to the user.

Reviewer 2: ```suggestion
featured_quote: Les feuilles de style en cascade (CSS) sont utilisées pour peindre, mettre en forme et organiser visuellement le contenu des pages Web. Leurs capacités couvrent des concepts aussi simples que la couleur du texte ou la perspective 3D. Elles disposent également de possibilités pour permettre aux développeurs de gérer différentes tailles d’écran, différents contextes de visualisation, ainsi que l’impression. CSS aide les développeurs et développeuses à gérer le contenu et à s’assurer qu’il s’adapte correctement aux personnes.
```

featured_stat_1: 5%
featured_stat_label_1: Pages using custom properties
featured_stat_2: 2%
featured_stat_label_2: Sites that use CSS Grid
featured_stat_3: 780

> featured_stat_label_3: Number of digits in largest Z-Index value

Reviewer: Sorry one more thing - need these labels translated.

Author: Done here: https://github.com/HTTPArchive/almanac.httparchive.org/pull/1519/commits/1dbe7e73571275f1ce94ee8c3a1321f5c637bc1a

Author: And enhaced just after ^^

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=2091525146&format=interactive"
  )
}}

Aucun gagnant clair ici, `max-width` et `min-width` sont presque à égalité.

### Impression et vocalisation

Les sites Web ressemblent à du papier numérique, non&nbsp;? En tant qu’utilisateurs et utilisatrices, il est généralement connu que vous pouvez simplement lancer l’impression à partir de votre navigateur et transformer ce contenu numérique en contenu physique. Un site Web n’est pas obligé de se changer pour ce cas d’utilisation, mais il le peut s’il le souhaite&nbsp;! Moins connue est la possibilité d’ajuster votre site Web dans le cas où il est lu par un outil ou un robot. Alors, à quelle fréquence ces fonctionnalités sont-elles exploitées&nbsp;?

{{ figure_markup(
  image="fig32.png",
  caption='Adoption des medias `all`, `print`, `screen` et `speech` pour les <i lang="en">media queries</i>.',

>   description=’Diagramme à barres montrant 35&nbsp;% des pages de bureau utilisent le type de requête `all`, 46&nbsp;% utilisent `print`, 72&nbsp;% utilisent `screen` et 0&nbsp;% utilisent `speech`. L’adoption est inférieure d’environ 5&nbsp;% pour les ordinateurs de bureau par rapport aux mobiles.’,

Reviewer: Incorrect quotes used:

```suggestion
  description='Diagramme à barres montrant 35&nbsp;% des pages de bureau utilisent le type de requête `all`, 46&nbsp;% utilisent `print`, 72&nbsp;% utilisent `screen` et 0&nbsp;% utilisent `speech`. L’adoption est inférieure d’environ 5&nbsp;% pour les ordinateurs de bureau par rapport aux mobiles.',
```

      <tr>
        <td>styles.css?ver=5.1.3</td>
        <td class="numeric">0.32&nbsp;%</td>
        <td class="numeric">0.35&nbsp;%</td>
      </tr>
      <tr>
        <td>custom.css</td>
        <td class="numeric">0.32&nbsp;%</td>
        <td class="numeric">0.33&nbsp;%</td>
      </tr>
      <tr>
        <td>print.css</td>
        <td class="numeric">0.32&nbsp;%</td>
        <td class="numeric">0.28&nbsp;%</td>
      </tr>
      <tr>
        <td>responsive.css</td>
        <td class="numeric">0.28&nbsp;%</td>
        <td class="numeric">0.31&nbsp;%</td>
      </tr>
    </tbody>
  </table>
  <figcaption>{{ figure_link(caption="Noms de feuilles de style les plus fréquemment utilisés.") }}</figcaption>
</figure>

Regardez tous ces noms de fichiers créatifs&nbsp;! `style`, `styles`, `main`, `default`, `all`. L’un se démarque cependant, le voyez-vous&nbsp;? `BfWyFJ2Rl5s.css` prend la quatrième place pour les plus populaires. Nous avons fait quelques recherches et notre meilleure hypothèse est qu’il est lié aux boutons «J’aime» de Facebook. Savez-vous quel est ce fichier&nbsp;? Laissez un commentaire, car nous aimerions entendre l’histoire.

### Poids de la feuille de style

> Quelle est le poids de ces feuilles de style&nbsp;? Est-ce que notre embonpoint CSS est quelque chose à craindre&nbsp;? À en juger par ces données, CSS n’est pas le principal coupable du gonflement des pages.

Reviewer 2: ```suggestion
Quel est le poids de ces feuilles de style&nbsp;? Est-ce que notre embonpoint CSS est quelque chose à craindre&nbsp;? À en juger par ces données, CSS n’est pas le principal coupable du gonflement des pages.
```

{{ figure_markup(
  image="fig35.png",
  caption="Distribution du nombre d’octets de feuille de style (Ko) chargés par page.",
  description="Distribution du nombre d’octets de feuilles de style chargés par page. Les 10e, 25e, 50e, 75e et 90e centiles de la page de bureau sont&nbsp;: 8, 26, 62, 129 et 240&nbsp;Ko. La distribution de bureau est légèrement supérieure à la distribution mobile de 5 à 10&nbsp;Ko.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=2132635319&format=interactive"
  )
}}

> Voir le chapitre [Poids des page](./page-weight) pour un aperçu plus détaillé du nombre d’octets que les sites Web chargent pour chaque type de contenu.

Reviewer 2: ```suggestion
Voir le chapitre [Poids des pages](./page-weight) pour un aperçu plus détaillé du nombre d’octets que les sites Web chargent pour chaque type de contenu.
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQO5CabwLwQ5Lj1_9bbEFnFM1qEqCorymaBHrcaNiMSJ7sYDKHUI5iish5VAS-SxN447UTW-1-5-OjE/pubchart?oid=1318910215&format=interactive"
  )
}}

Il s’avère qu’environ un tiers du Web utilise [`normalize.css`](https://necolas.github.io/normalize.css), ce qui pourrait être considéré comme une approche douce, en tout cas plus que celle d’un pur <i lang="en">reset</i> CSS. Nous avons regardé un peu plus en profondeur, et il s’avère que <span lang="en">Bootstrap</span> inclut `normalize.css`, ce qui représente probablement une grande partie de son utilisation. Il convient également de noter que `normalize.css` a plus d’adoption que <span lang="en">Bootstrap</span>, donc il y a beaucoup de gens qui l’utilisent seul.

### `@supports` et `@import`

CSS `@supports` est un moyen pour le navigateur de vérifier si une combinaison propriété-valeur particulière est analysée comme valide, puis d’appliquer des styles si la vérification est vraie.

{{ figure_markup(
  image="fig38.png",

>   caption='Popularité des règles CSS «@».",

Reviewer: Incorrect quote used:

```suggestion
  caption="Popularité des règles CSS «@».",
```
